xwiki-pro-macros-api/src/main/java/com/xwiki/macros/tab/internal/TabGroupMacro.java

```diff
@@ -0,0 +1,253 @@
+
+        Block contentBlocks = this.contentParser.parse(content, context, false, context.isInline());
+
+        List<Block> ulBlocks = new ArrayList<>();
+        List<Block> macroBlocks = contentBlocks.getBlocks(new MacroBlockMatcher(TAB_MACRO_ID), Block.Axes.CHILD);
+        Optional<Block> defaultTabBlock = macroBlocks.stream()
+            .filter(i -> TRUE.equals(i.getParameter(TabMacroParameters.PARAM_NAME_DEFAULT)))
+            .findFirst();
+        int defaultTabIndex = defaultTabBlock.isPresent() ? macroBlocks.indexOf(defaultTabBlock.get()) : 0;
+
+        int inc = 0;
+        for (Block mb : macroBlocks) {
+            boolean isActive = inc == defaultTabIndex;
+            mb.setParameter(TabMacroParameters.PARAM_NAME_DEFAULT, isActive ? TRUE : FALSE);
+            String id = mb.getParameter(ID);
+            if (id == null) {
+                id = macroId + "_" + inc;
+                mb.setParameter(ID, id);
+            }
+            // Set animation parameter for sub macro block
+            if (parameters.getNextAfter() != 0 && mb.getParameter(PARAM_NAME_NEXT_AFTER) == null) {
+                mb.setParameter(PARAM_NAME_NEXT_AFTER, Integer.toString(parameters.getNextAfter()));
+            }
+            if (parameters.getEffect() != null && mb.getParameter(PARAM_NAME_EFFECT) == null) {
+                mb.setParameter(PARAM_NAME_EFFECT, parameters.getEffect().name());
+            }
+            if (parameters.getEffectDuration() != 0 && mb.getParameter(PARAM_NAME_EFFECT_DURATION) == null) {
+                mb.setParameter(PARAM_NAME_EFFECT_DURATION, Integer.toString(parameters.getEffectDuration()));
+            }
+
+            Map<String, String> lbParam = new HashMap<>();
+            lbParam.put(BLOCK_PARAM_ROLE, "presentation");
+            if (isActive) {
+                lbParam.put(BLOCK_PARAM_CLASS, "active");
+            }
+            // We use the raw block because the LinkBlock generate a span element which break bootstrap tabs CSS
+            RawBlock linkBlock = new RawBlock(
+                String.format("<a href=\"#%s\" aria-controls=\"%s\" role=\"tab\" data-toggle=\"tab\">%s</a>",
+                    id, id, XMLUtils.escape(mb.getParameter(TAB_MACRO_PARAM_LABEL))),
+                Syntax.HTML_5_0);
+            ulBlocks.add(new ListItemBlock(Collections.singletonList(linkBlock), lbParam));
+            inc++;
+        }
+        Map<String, String> mainBlockParam = new HashMap<>();
+        StringBuilder style = new StringBuilder();
+        StringBuilder mainDivClass = new StringBuilder("xwikitabmacro");
+        if (!StringUtils.isEmpty(parameters.getId())) {
+            mainBlockParam.put(ID, parameters.getId());
+        }
+        if (!StringUtils.isEmpty(parameters.getWidth())) {
+            style.append("width: ").append(parameters.getWidth());
+        }
+        if (!StringUtils.isEmpty(parameters.getHeight())) {
+            style.append("height: ").append(parameters.getWidth());
```

Comment:
This looks like a typo, shouldn't this be

```suggestion
            style.append("height: ").append(parameters.getHeight());
```